DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/igmp/test_igmp_snooping.py

```diff
@@ -0,0 +1,482 @@
+
+    router_entires = out[0][device_name]['parsed_output'][0]['router']
+    mdb_entires = out[0][device_name]['parsed_output'][0]['mdb']
+    return mdb_entires, router_entires
+
+
+async def common_bridge_and_igmp_setup(device_name, bridge, igmp_ver, dut_ports, querier_interval=0):
+    """
+    Common setup part for bridge and igmp
+    Args:
+        device_name (str): Dent device host name
+        bridge (str): Bridge device name
+        igm_ver (int): IGMP Version to set on bridge dev
+        dut_ports (list): List containing Dent device ports
+        querier_interval (int): Multicast querier_interval to be set
+    """
+
+    out = await IpLink.add(
+        input_data=[{device_name: [
+            {'device': bridge, 'type': 'bridge', 'vlan_filtering': 1}]}])
+    err_msg = f'Verify that bridge created and vlan filtering set successful\n{out}'
+    assert out[0][device_name]['rc'] == 0, err_msg
+
+    out = await IpLink.set(
+        input_data=[{device_name: [
+            {'device': bridge, 'operstate': 'up'}]}])
+    err_msg = f"Verify that bridge set to 'UP' state.\n{out}"
+    assert out[0][device_name]['rc'] == 0, err_msg
+
+    out = await IpLink.set(
+        input_data=[{device_name: [
+            {'device': bridge, 'type': 'bridge', 'mcast_snooping': 1, 'mcast_igmp_version': igmp_ver}]}])
+    err_msg = f'Verify that bridge set mcast_snooping 1 and mcast_igmp_version {igmp_ver}.\n{out}'
+    assert out[0][device_name]['rc'] == 0, err_msg
+
+    if querier_interval:
+        out = await IpLink.set(
+            input_data=[{device_name: [
+                {'device': bridge, 'type': 'bridge', 'mcast_querier': 1,
+                 'mcast_querier_interval': querier_interval * 100}]}])
+        err_msg = f'Verify bridge set mcast_querier 1 and mcast_querier_interval to {querier_interval * 100}.\n{out}'
+        assert out[0][device_name]['rc'] == 0, err_msg
+
+    out = await IpLink.set(
+        input_data=[{device_name: [
+            {'device': port, 'master': bridge, 'operstate': 'up'} for port in dut_ports]}])
+    err_msg = f"Verify that bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
+    assert out[0][device_name]['rc'] == 0, err_msg
+
+
+def verify_expected_rx_rate(stats, tx_port, rx_ports, rate_coef=0, deviation=0.10):
+    """
+    Verify expected rx_rate in bytes on ports
+    Args:
+        stats (stats_object): Output of tgen_utils_get_traffic_stats
+        tx_port (str): Name of Tx Port
+        rx_ports (list): list of Rx ports which rate should be verified
+        rate_coef (float): Rate coefficient for expected rx_rate
+        deviation (int): Permissible deviation percentage
+    """
+    collected = {row['Port Name']:
+                 {'tx_rate': row['Bytes Tx. Rate'], 'rx_rate': row['Bytes Rx. Rate']} for row in stats.Rows}
+    tx_name = tx_port.split('_')[0]
+    exp_rate = float(collected[tx_name]['tx_rate']) * rate_coef if rate_coef else float(collected[tx_name]['tx_rate'])
```

Comment:
Make rate_coef default to 1 and simply do:
`exp_rate = float(collected[tx_name]['tx_rate']) * rate_coef`